static/app/views/issueList/issueViews/issueViewsList/issueViewsList.tsx

```diff
@@ -78,13 +80,39 @@ function IssueViewSection({createdBy, limit, cursorQueryParam}: IssueViewSection
           view.id === variables.id ? {...view, starred: variables.starred} : view
         );
       });
+      if (variables.starred) {
+        setApiQueryData<StarredGroupSearchView[]>(
+          queryClient,
+          makeFetchStarredGroupSearchViewsKey({orgSlug: organization.slug}),
+          data => [...(data ?? []), variables.view]
+        );
+      } else {
+        setApiQueryData<StarredGroupSearchView[]>(
+          queryClient,
+          makeFetchStarredGroupSearchViewsKey({orgSlug: organization.slug}),
+          data => data?.filter(view => view.id !== variables.id)
+        );
+      }
     },
     onError: (_error, variables) => {
       setApiQueryData<GroupSearchView[]>(queryClient, tableQueryKey, data => {
         return data?.map(view =>
           view.id === variables.id ? {...view, starred: !variables.starred} : view
         );
       });
+      if (variables.starred) {
+        setApiQueryData<StarredGroupSearchView[]>(
+          queryClient,
+          makeFetchStarredGroupSearchViewsKey({orgSlug: organization.slug}),
+          data => data?.filter(view => view.id !== variables.id)
+        );
+      } else {
+        setApiQueryData<StarredGroupSearchView[]>(
+          queryClient,
+          makeFetchStarredGroupSearchViewsKey({orgSlug: organization.slug}),
+          data => [...(data ?? []), variables.view]
```

Comment:
If there was an error unstarring your first view, this would put it back on the bottom which isn't right. 

Because we don't know where the starred view should have been, it might be easier to omit the newly added `setApiQueryData`s here. The `useUpdateGroupSearchViewStarred` hook already invalidates the starred views query on success/error so it will update, just not immediately. IMO that's fine since it's not directly related to the thing the user clicked on (which should update immediately).

Kind of an edge case so I'll leave it up to you whether you want to keep this or not